src/clr-angular/data/datagrid/datagrid-detail.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+  Component,
+  Directive,
+  EventEmitter,
+  Input,
+  TemplateRef,
+  ViewContainerRef,
+  Output,
+  OnInit,
+  OnDestroy,
+  ViewChild,
+  HostListener,
+} from '@angular/core';
+import { DetailService } from './providers/detail.service';
+import { Subscription } from 'rxjs';
+import { FocusTrapDirective } from '../../utils/focus-trap/focus-trap.directive';
+
+@Component({
+  selector: 'clr-dg-detail',
+  host: {
+    '[class.datagrid-detail-pane-wrapper]': 'true',
+  },
+  template: `
+    <div clrFocusTrap class="datagrid-detail-pane">
+      <ng-content></ng-content>
+    </div>
+    `,
+})
+export class ClrDatagridDetail {
+  constructor(private detailService: DetailService) {}
+
+  @HostListener('body:keyup.escape')
+  close() {
+    this.detailService.close();
+  }
+}
+
+@Directive({
+  selector: '[clrIfDetail]',
+})
+export class ClrDatagridDetailDirective implements OnInit, OnDestroy {
```

Comment:
We need the structural directive to pass in the data for the selected row, so unless the detail pane is static content it would not work without it. 

I don't see a reuse of this conditional directive anywhere.